orchestration/src/main/java/com/sap/ai/sdk/orchestration/ConfigToRequestTransformer.java

```diff
@@ -42,14 +43,14 @@ static TemplatingModuleConfig toTemplateModuleConfig(
      * In this case, the request will fail, since the templating module will try to resolve the parameter.
      * To be fixed with https://github.tools.sap/AI/llm-orchestration/issues/662
      */
-    val messages = Option.of(template).map(TemplatingModuleConfig::getTemplate).getOrElse(List::of);
+    val messages = Option.of(template).map(t -> ((Template) t).getTemplate()).getOrElse(List::of);
```

Comment:
Not sure why the compiler doesn't complain about an unchecked cast here..

```suggestion
    val messages =
        Option.of(template)
            .filter(Template.class::isInstance)
            .map(Template.class::cast)
            .map(Template::getTemplate)
            .getOrElse(List::of);
```

Reply:
> why

I assume, it's because the interface has only one known implementing class.